@here/harp-datasource-protocol/lib/Techniques.ts

```diff
@@ -1089,8 +962,166 @@ export interface TextTechnique extends BaseTechnique {
+}
+
+/**
+ * Runtime representation of [[CirclesStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface CirclesTechnique extends PointTechniqueParams {
+    name: "circles";
 }
 
+/**
+ * Runtime representation of [[PoiStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface PoiTechnique extends MarkerTechniqueParams {
+    name: "labeled-icon";
+}
+
+/**
+ * Runtime representation of [[LineMarkerStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface LineMarkerTechnique extends MarkerTechniqueParams {
+    name: "line-marker";
+}
+
+/**
+ * Runtime representation of [[SegmentsStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface SegmentsTechnique extends SegmentsTechniqueParams {
+    name: "segments";
+}
+
+/**
+ * Runtime representation of [[BasicExtrudedLineStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface BasicExtrudedLineTechnique extends BasicExtrudedLineTechniqueParams {
+    name: "extruded-line";
+}
+
+/**
+ * Runtime representation of [[StandardExtrudedLineStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface StandardExtrudedLineTechnique extends StandardExtrudedLineTechniqueParams {
+    name: "extruded-line";
+}
+
+/**
+ * Runtime representation of [[SolidLineStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface SolidLineTechnique extends SolidLineTechniqueParams {
+    name: "solid-line";
+}
+
+/**
+ * Runtime representation of [[DashedLineStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface DashedLineTechnique extends DashedLineTechniqueParams {
+    name: "dashed-line";
+}
+
+/**
+ * Runtime representation of [[LineStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface LineTechnique extends LineTechniqueParams {
+    name: "line";
+}
+
+/**
+ * Runtime representation of [[FillStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface FillTechnique extends FillTechniqueParams {
+    name: "fill";
+}
+
+/**
+ * Runtime representation of [[ExtrudedPolygonStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface ExtrudedPolygonTechnique extends ExtrudedPolygonTechniqueParams {
+    name: "extruded-polygon";
+}
+
+/**
+ * Runtime representation of [[TextStyle]] as parsed by [[StyleSetEvaluator]].
+ */
+export interface TextTechnique extends TextTechniqueParams {
+    name: "text";
+}
+
+/**
+ * Technique used to render a mesh geometry.
+ */
+export interface StandardTechnique extends BaseStandardTechniqueParams {
+    name: "standard";
+}
+
+export interface ShaderTechnique extends ShaderTechniqueParams {
+    /**
+     * Name of technique. Is used in the theme file.
+     */
+    name: "shader";
+}
+
+/**
+ * Technique used to render a mesh geometry with textures.
+ */
+export interface StandardTexturedTechnique extends StandardTexturedTechniqueParams {
+    name: "standard-textured";
+}
+
+/**
+ * Possible techniques that can be used to draw a geometry on the map.
+ */
+export type Technique =
```

Comment:
This is really ugly and not extendable. Users that implement a datasource can not implement their own techniques. Instead we should have some interface for techniques (or separate techniques into geometry processing and materials), but I guess this will come with Robertos style rework.

Reply:
Agree, but you either have closed / properly typing or open interface.
It was always closed, and I choose to leave it as is as we have no clear vision where to go ...

Custom technique would require extension points in so many places, that it's rather separate task.